script/search/search.inc.php

```diff
@@ -0,0 +1,335 @@
+      $result->isomatch = false;
+      $result->allmatch = false;
+      $result->regionmatch = false;
+      $result->legacy = false;
+
+      $result->textparts = explode(':', $query);
+
+      for($i = 0; $i < sizeof($result->textparts) - 1; $i++) {
+        $match = strtolower($result->textparts[$i]);
+        if(!strcmp($match, 'id')) $result->isomatch = true;
+        else if(!strcmp($match, 'iso')) $result->isomatch = true;
+        else if(!strcmp($match, 'all')) $result->allmatch = true;
+        else if(!strcmp($match, 'region')) $result->regionmatch = true;
+        else if(!strcmp($match, 'legacy')) $result->legacy = true;
+        else if(!strncmp($match, 'country', strlen($match))) $result->rangematch = rmCountry;
+        else if(!strncmp($match, 'language', strlen($match))) $result->rangematch = rmLanguage;
+        else if(!strncmp($match, 'keyboard', strlen($match))) $result->rangematch = rmKeyboard;
+      }
+
+      if($result->isomatch && $result->rangematch == rmAll) {
+        // Don't support match on ID for all contexts
+        $result->isomatch = false;
+      }
+
+      $result->text = array_pop($result->textparts);
+      $result->searchtext = $result->text;
+
+      //$result->text = str_replace('*', '%', $result->text);
+      //$result->text = str_replace('_', '?', $result->text);
+
+      if($result->regionmatch) {
+        /*$r = new CRM_Region();
+        if(!$r->Load($result->searchtext)) $result->region='Unknown';
+        else $result->region=$r->Name;*/
+        $result->region='Unknown';
+      }
+
+      $this->result = $result;
+      return true;
+    }
+
+    function GetSearchText() {
+      $this->GetSearchQueries($rangetext, $count, $countries, $langs, $keyboards);
+      return $this->result->searchtext;
+    }
+
+    private $gsq_complete=false, $gsq_rangetext, $gsq_count, $gsq_countries = [], $gsq_langs = [], $gsq_keyboards = [];
+
+    private function GetSearchQueries(&$rangetext, &$count, &$countries, &$langs, &$keyboards) {
+      if(!$this->gsq_complete) {
+        $this->result->searchtext = strip_tags($this->result->searchtext);
+        switch($this->result->rangematch) {
+          case rmAll:
+            $this->gsq_rangetext = ''; break;
+            //$rangetext = "Matches found for '{$this->result->searchtext}'"; break;
+          case rmKeyboard:
+            if($this->result->isomatch) $this->gsq_rangetext = "Keyboard with id '{$this->result->searchtext}'";
+            else if($this->result->legacy) $this->gsq_rangetext = "Keyboard with legacy id '{$this->result->searchtext}'";
+            else $this->gsq_rangetext = "Keyboards matching '{$this->result->searchtext}'"; break;
+          case rmLanguage:
+            if($this->result->allmatch) $this->gsq_rangetext = "All languages matching '{$this->result->searchtext}'";
+            else if($this->result->isomatch) $this->gsq_rangetext = "Keyboards for language with BCP 47 code '{$this->result->searchtext}'";
```

Comment:
Is `result->isomatch` used interchangeably with BCP 47 and ISO3166-1 (l.103)?

Reply:
Yes. Although I will be reworking some of this in a future PR.